modules/os2forms_fbs_handler/src/Client/FBS.php

```diff
@@ -0,0 +1,234 @@
+   * Check if user exists.
+   *
+   * @param string $cpr
+   *   The users personal security number.
+   *
+   * @return \Drupal\os2forms_fbs_handler\Client\Model\Patron|null
+   *   NULL if not else the Patron.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   * @throws \JsonException
+   */
+  public function doUserExists(string $cpr): ?Patron {
+    // Check if session have been created with FBS and if not create it.
+    if (!$this->isLoggedIn()) {
+      $this->login();
+    }
+
+    // Try pre-authenticate the user/parent.
+    $json = $this->request('/external/{agency_id}/patrons/preauthenticated/v9', $cpr);
+    if ($json->authenticateStatus === 'VALID') {
+      return new Patron(
+        $json->patron->patronId,
+        (bool) $json->patron->receiveSms,
+        (bool) $json->patron->receivePostalMail,
+        $json->patron->notificationProtocols,
+        $json->patron->phoneNumber,
+        is_null($json->patron->onHold) ? $json->patron->onHold : (array) $json->patron->onHold,
+        $json->patron->preferredLanguage,
+        (bool) $json->patron->guardianVisibility,
+        $json->patron->emailAddress,
+        (bool) $json->patron->receiveEmail,
+        $json->patron->preferredPickupBranch
+      );
+    }
+
+    return NULL;
+  }
+
+  /**
+   * Create new patron with guardian attached.
+   *
+   * @param \Drupal\os2forms_fbs_handler\Client\Model\Patron $patron
+   *   The patron to create.
+   * @param \Drupal\os2forms_fbs_handler\Client\Model\Guardian $guardian
+   *   The guardian to attach to the parton.
+   *
+   * @return mixed
+   *   JSON response from FBS.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   * @throws \JsonException
+   */
+  public function createPatronWithGuardian(Patron $patron, Guardian $guardian) {
+    $uri = '/external/{agency_id}/patrons/withGuardian/v1';
+    $payload = [
+      'cprNumber' => $patron->cpr,
+      'pincode' => $patron->pincode,
+      'preferredPickupBranch' => $patron->preferredPickupBranch,
+      'name' => 'Unknown Name',
+      'email' => $patron->emailAddress,
+      'guardian' => $guardian->toArray(),
+    ];
+
+    return $this->request($uri, $payload,);
+  }
+
+  /**
+   * Update patron information.
+   *
+   * @param \Drupal\os2forms_fbs_handler\Client\Model\Patron $patron
+   *   The patron to update.
+   *
+   * @return bool
+   *   TRUE if success else FALSE.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   * @throws \JsonException
+   */
+  public function updatePatron(Patron $patron): bool {
+    $uri = '/external/{agency_id}/patrons/' . $patron->patronId . '/v6';
+    $payload = [
+      'patronid' => $patron->patronId,
+      'patron' => $patron->toArray(),
+      'pincodeChange' => [
+        'pincode' => $patron->pincode,
+        'libraryCardNumber' => $patron->cpr,
+      ],
+    ];
+
+    $json = $this->request($uri, $payload, RequestMethodInterface::METHOD_PUT);
+
+    return $json->authenticateStatus === 'VALID';
```

Comment:
suggesting to create a class const for this word **VALID**. It has beeb used more than twice already.

Reply:
Fixed